docs/readmes/proposals/p018_ue_requested_pdu_session_modification.md

```diff
@@ -0,0 +1,205 @@
+   procedure after expiry of timer T3396. 
+
+4. When the UE-requested PDU session modification procedure is used to indicate
+   to request specific QoS handling and segregation of service data flows 
+
+## **Use Case**
+
+The UE can request the network to modify or release PDU sessions. The network
+can fulfill such a request from the UE by modifying a PDU session or releasing
+a PDU session using network-requested procedures.  
+
+When a UE establishes a PDU session to a DN, a default non-GBR (non-Guaranteed
+Bit Rate) QoS Flow without any Packet Filter is also formed to carry the UE’s
+traffic to the DN and vice versa. The UE or AFs (Application Functions) can
+create additional QoS Flows with different QoS characteristics using the PDU
+Session Modification procedures. For example, the UE can initiate a PDU Session
+Modification to create a new GBR QoS Flow for a video conference session with
+criteria such as: 
+
+- UDP 	port = 1234 
+
+- latency < 10ms 
+
+- UL/DL bandwidth = 2 Mbps GBR, Etc… 
+
+## **State changes in UE and PDU session**
+
+![UE and PDU session states](../assets/proposals/p018/UE_and_PDU_session_states.png)
+
+## **Call Flows**
+
+![Sequence diagram for UE requested PDU session modification](../assets/proposals/p018/UE_req_PDU_sess_modification.png)
+
+## **Proposal**
+
+### **Detailed Design**
+
+1.a. UE initiates the PDU Session Modification procedure by the transmission of
+     a NAS message. The NAS message is forwarded by the RAN to the AMF with an
+     indication of User location Information and AMF invokes API
+     Nsmf_PDUSession_UpdateSMContext to SMF. 
+
+1.b. RAN shall indicate to the SMF when the AN resource onto which a QoS Flow is
+     mapped are released irrespective of whether notification control is
+     configured. RAN sends the N2 message with SM information includes the QFI,
+     User location Information and an indication that the QoS Flow is released.
+     The AMF invokes Nsmf_PDUSession_UpdateSMContext  
+
+     Below are the N1 SM container contents: 
+
+     -  N1 SM container 
+
+        --  PDU Session Modification Request  
+
+             ---  PDU session ID,  
+
+             ---  Packet Filters, 
+
+             ---  Operation, 
+
+             ---  Requested QoS, 
+
+             ---  Segregation, 
+
+             ---  5GSM Core Network Capability,  
+
+             ---  Number Of Packet Filters,  
+
+             ---  [Always-on PDU Session Requested],  
+
+       --  UE Integrity Protection Maximum Data Rate 
+
+     The NAS message is forwarded by (R)AN to AMF with indication of user
+     location information. AMF calls Nsmf_PDUSession_UpdateSMContext (SM context
+     ID, N1 SM container (PDU session modification request)). 
+
+2. SMF responds AMF after updating SM_Context and SMF requests to transfer
+   downlink N2 message to the UE and AN through the AMF. i.e. The SMF request
+   to update activation of Data Radio Bearer and N3 tunnel for the PDU session. 
+
```

Comment:
When SMF receives the request to modify Qos, how does SMF decides if this change is valid or not? 
Does SMF stores any kind of profile information ? Does SMF do a query to any other system to check if the Qos change is allowed?